File: 1004. Max Consecutive Ones III/Solution.cs
public class Solution {
    public int LongestOnes(int[] nums, int k) {

        int maxValidSequenceLength = 0, sequenceZeros = 0;
        for (int l = 0, r = 0; r < nums.Length; ++r) {
            
            if (nums[r] == 0) sequenceZeros++;

            while (sequenceZeros > k) {
                if (nums[l++] == 0) sequenceZeros--;
            }

            maxValidSequenceLength = Math.Max(maxValidSequenceLength, r-l+1);
        }
        return maxValidSequenceLength;
    }
}

File: 1004. Max Consecutive Ones III/solution.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:

        maxValidSequenceLength, sequenceZeros, l = 0, 0, 0
        for r, num in enumerate(nums):

            if num == 0:
                sequenceZeros += 1

            while sequenceZeros > k:
                if nums[l] == 0:
                    sequenceZeros -= 1
                l += 1    

            maxValidSequenceLength = max(maxValidSequenceLength, r-l+1)
        
        return maxValidSequenceLength
